src/components/horizontal_rule/horizontal_rule.tsx

```diff
@@ -1,7 +1,20 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { SFC, HTMLAttributes } from 'react';
 import classNames from 'classnames';
 
+import { CommonProps } from '../common';
+
+export type EuiHorizontalRuleSize = 'full' | 'half' | 'quarter';
```

Comment:
To avoid duplicating these sizes with the `sizeToClassNameMap` further down, you can bring sizeToClassNameMap above this line and change this to

```suggestion
export type EuiHorizontalRuleSize = keyof typeof sizeToClassNameMap;
```

Same thing with the margins.

(props to @jasonrhodes for finding this)